shiny/ui/_chat.py

```diff
@@ -1107,16 +1107,13 @@ def update_user_input(
             }
         )
 
-        _utils.run_coro_sync(
-            self._session.send_custom_message(
-                "shinyChatMessage",
-                {
-                    "id": self.id,
-                    "handler": "shiny-chat-update-user-input",
-                    "obj": obj,
-                },
-            )
-        )
+        msg = {
+            "id": self.id,
+            "handler": "shiny-chat-update-user-input",
+            "obj": obj,
+        }
+
+        self._session._send_message_sync({"custom": {"shinyChatMessage": msg}})
```

Comment:
This is def an improvement over `.run_coro_sync()`, but I'm wondering if we should make `.update_user_input()` async and escape the possible internal dangers of `._send_message_sync()`.

Reply:
Thanks, we pretty intentionally wanted to try avoiding async since it’s so useful to provide a default value at the top-level.

That said, if it becomes a problem, we could probably add `value` to `.ui()` to help with that issue.